portfolio/src/main/java/com/google/sps/servlets/ViewPuzzlePageServlet.java

```diff
@@ -14,14 +14,21 @@
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
     response.setContentType("text/html;");
 
+    
+    String id = request.getParameter("id");
+    if(id == null) {
+      //if an id was not given
+      response.sendRedirect("viewpuzzle.html");
```

Comment:
The UI really can't do much without the puzzle so I would recommend also returning an error message in this case:
```suggestion
    response.setStatus(HttpServletResponse.SC_BAD_REQUEST);
    response.getWriter().println("The id parameter is required");
```

Reply:
it redirects because the navbar uses _/viewpuzzle_ and _/createpuzzle._
![navbar](https://user-images.githubusercontent.com/33354499/115048130-daee3e00-9e9e-11eb-9788-3d93be8d8fff.png)

if the error message is added the user will get this when clicking on solve.
![error5](https://user-images.githubusercontent.com/33354499/115048177-e80b2d00-9e9e-11eb-99dd-ed4b84323864.png)

Reply:
I see, in that case we should change that link to point to the ListPuzzlePage for issue #22. But that can be done in a separate PR.